kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ResourceSelection.vue

```diff
@@ -382,57 +399,69 @@
           !selectPracticeQuiz.value &&
-          contentList.value.every(content => folderDoesNotHaveTooManyQuestions(content))
+          contentList.value.every(content => nodeIsSelectableOrUnselectable(content)) &&
+          // We only show the select all button if both all the checkboxes are enabled,
+          // and adding all the currently unselected questions in resources/folders plus
+          // those that are already selected (both in this level of the topic tree and elsewhere)
+          // would not exceed the maxSectionQuestionOptions.
+          // Do this by taking away the selected questions from the total questions in the
+          // resource/folder and checking if the sum of all of these is less than or equal to
+          // the remaining number of questions that can be added.
+          contentList.value.reduce(
+            (acc, content) =>
+              unusedQuestionsCount(content) - selectedQuestionsFromNode(content) + acc,
+            0,
+          ) <=
+          maxSectionQuestionOptions.value - workingPoolUnusedQuestions.value
         );
       });
 
-      function handleSelectAll(isChecked) {
-        if (isChecked) {
-          addToWorkingResourcePool(selectableContentList());
-        } else {
-          contentList.value.forEach(content => {
-            var contentToRemove = [];
-            if (content.kind === ContentNodeKinds.TOPIC) {
-              contentToRemove = content.children.results;
-            } else {
-              contentToRemove.push(content);
-            }
-            contentToRemove.forEach(c => {
-              removeFromWorkingResourcePool(c);
-            });
-          });
-        }
-      }
-
       /**
        * @param {Object} param
        * @param {ContentNode} param.content
        * @param {boolean} param.checked
        * @affects workingResourcePool - Adds or removes the content from the workingResourcePool
        * When given a topic, it adds or removes all the exercises in the topic from the
-       * workingResourcePool. This assumes that topics which should not be added are not able to
-       * be checked and does not do any additional checks.
+       * workingResourcePool.
        */
-      function toggleSelected({ content, checked }) {
-        content = content.kind === ContentNodeKinds.TOPIC ? content.children.results : [content];
+      async function toggleSelected({ content, checked }) {
+        if (content.is_leaf) {
+          content = [content];
+        } else {
+          // If we already have all of the children locally, and every child is a leaf, we can
+          // just add them all to the working resource pool
+          if (!content.children.more && !content.children.results.some(n => !n.is_leaf)) {
+            content = content.children.results;
+          } else {
+            // If we don't have all of the children locally, we need to fetch them
+            const children = await ContentNodeResource.fetchCollection({
```

Comment:
Haven't tested yet - but just wondering what this might look like on a slow connection and if we might want to add a loading state here.

Reply:
With that said, we probably need to work on adding loading states in various places... 


https://github.com/learningequality/kolibri/assets/6356129/43a8f348-f054-4ebf-972a-a605ebe9bf14